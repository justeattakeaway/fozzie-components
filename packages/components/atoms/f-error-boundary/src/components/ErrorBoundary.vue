<script>
import loggerService from '../service/logger.service';

export default {
    props: {
        tier: {
            type: String,
            default: null
        },

        stopPropagation: {
            type: Boolean,
            default: true
        },

        hideOnError: {
            type: Boolean,
            default: false
        }
    },

    data: () => ({
        hasError: false,
        error: null,
        vm: null,
        info: null
    }),

    computed: {
        loggerPayload () {
            return loggerService.buildVueError(this.error, this.vm, this.info, {
                tier: this.tier
            });
        }
    },

    errorCaptured (error, vm, info) {
        return this.interceptError(error, vm, info);
    },

    methods: {
        interceptError (error, vm, info) {
            this.hasError = true;
            this.error = error;
            this.vm = vm;
            this.info = info;

            this.$emit('on-error', {
                error,
                vm,
                info,
                tier: this.tier,
                loggerPayload: this.loggerPayload
            });

            return !this.stopPropagation;
        }
    },

    render () {
        return this.hideOnError && this.hasError
            ? null
            : this.$scopedSlots.default?.({
                hasError: this.hasError,
                error: this.error,
                vm: this.vm,
                info: this.info,
                tier: this.tier,
                loggerPayload: this.loggerPayload
            });
    }
};
</script>
