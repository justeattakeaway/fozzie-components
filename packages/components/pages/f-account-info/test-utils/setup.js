import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import { VueI18n } from '@justeat/f-globalisation';
import tenantConfigs from '../src/tenants';

export const localVue = createLocalVue();
localVue.use(VueI18n);
localVue.use(Vuex);

export const i18n = {
    locale: 'en-GB',
    messages: {
        'en-GB': tenantConfigs['en-GB'].messages
    }
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const consumerDetailsGetResponse = {
    data: {
        ConsumerId: 1050450174,
        Id: '8It2lEaJjG82nJtxO00Iab0CoF0=',
        CreatedDate: '2019-04-16T17:25:00.000Z',
        EmailAddress: 'satchin.moriyoka@example.com',
        FirstName: 'Satchin',
        LastName: 'Moriyoka',
        PhoneNumber: '+441234567890',
        DateOfBirth: '1914-04-16T17:25:00.000Z',
        TermsAndConditionsId: 0,
        WantsNewsletter: true,
        AccountType: 'Default',
        ConsumerStatus: 'ReturnConsumer'
    }
};

const consumerAddressGetResponse = {
    AddressId: 1050450174,
    City: 'Bristol',
    ZipCode: 'BS1 4DJ',
    AddressName: 'Office',
    IsDefault: true, // Key field
    Line1: 'Just Eat',
    Line2: 'Broad Quay House'
};

export const consumerAddressesGetResponse = {
    data: [{
        ...consumerAddressGetResponse
    },
    {
        AddressId: 1050450175,
        City: 'Bristol',
        ZipCode: 'BS1 9DJ',
        AddressName: 'Office',
        IsDefault: false,
        Line1: 'Takeaway',
        Line2: 'Broad Quay House'
    }]
};

export const consumerDetailsMappedModel = {
    id: consumerDetailsGetResponse.data.ConsumerId,
    firstName: consumerDetailsGetResponse.data.FirstName,
    lastName: consumerDetailsGetResponse.data.LastName,
    phoneNumber: consumerDetailsGetResponse.data.PhoneNumber,
    emailAddress: consumerDetailsGetResponse.data.EmailAddress
};

export const consumerAddressMappedModel = {
    line1: consumerAddressGetResponse.Line1,
    line2: consumerAddressGetResponse.Line2,
    line3: consumerAddressGetResponse.Line3,
    locality: consumerAddressGetResponse.City,
    postcode: consumerAddressGetResponse.ZipCode
};

export const consumerViewModel = {
    ...consumerDetailsMappedModel,
    ...consumerAddressMappedModel
};

export const consumerStateModel = {
    consumer: {
        ...consumerDetailsMappedModel,
        ...consumerAddressMappedModel
    }
};
