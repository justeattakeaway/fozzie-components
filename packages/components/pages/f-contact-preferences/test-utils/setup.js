export const contactPreferencesGetResponse = {
    data: {
        Preferences: [{
            DisplayName: 'Order status',
            Sort: 1,
            Key: 'orderstatus',
            Push: false,
            Email: true,
            Sms: true
        }, {
            DisplayName: 'Review meal',
            Sort: 2,
            Key: 'reviewmeal',
            Push: false,
            Email: false,
            Sms: false
        }, {
            DisplayName: 'News & offers',
            Sort: 3,
            Key: 'news',
            Push: false,
            Email: false,
            Sms: false
        }
        ],
        PreferenceVersionViewed: 0
    }
};

export const contactPreferencesViewModel = {
    preferenceVersionViewed: 0,
    preferences: [{
        displayName: 'Order status',
        isEmailEnabled: false,
        emailValue: true,
        key: 'orderstatus',
        pushEnabled: false,
        pushValue: false,
        isSmsEnabled: false,
        smsValue: true,
        sort: 1,
        visible: true
    }, {
        displayName: 'Review meal',
        isEmailEnabled: true,
        emailValue: false,
        key: 'reviewmeal',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 2,
        visible: false
    }, {
        displayName: 'News & offers',
        isEmailEnabled: true,
        emailValue: false,
        key: 'news',
        pushEnabled: true,
        pushValue: false,
        isSmsEnabled: true,
        smsValue: false,
        sort: 3,
        visible: true
    }]
};

export const filteredContactPreferencesModel = [{
    displayName: 'Order status',
    isEmailEnabled: false,
    emailValue: true,
    key: 'orderstatus',
    pushEnabled: false,
    pushValue: false,
    isSmsEnabled: false,
    smsValue: true,
    sort: 1,
    visible: true
}, {
    displayName: 'News & offers',
    isEmailEnabled: true,
    emailValue: false,
    key: 'news',
    pushEnabled: true,
    pushValue: false,
    isSmsEnabled: true,
    smsValue: false,
    sort: 3,
    visible: true
}];

export const contactPreferencesUpdateModel = {
    Preferences: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: false,
        Sms: false
    }],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};

export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const token = 'some-auth-token';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

export const contactPreferencesUpdateBody = {
    Preference: [{
        DisplayName: 'Order status',
        Sort: 1,
        Key: 'orderstatus',
        Push: false,
        Email: true,
        Sms: true
    }, {
        DisplayName: 'Review meal',
        Sort: 2,
        Key: 'reviewmeal',
        Push: false,
        Email: false,
        Sms: false
    }, {
        DisplayName: 'News & offers',
        Sort: 3,
        Key: 'news',
        Push: false,
        Email: true,
        Sms: false
    }
    ],
    DeviceToken: null,
    DeviceType: null,
    PhoneNumber: null,
    PreferenceVersionViewed: 0
};
