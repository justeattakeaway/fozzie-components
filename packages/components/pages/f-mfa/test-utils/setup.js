export const baseUrl = 'https://smartGatewayBaseUrl.com';

export const conversationId = 'b7386108-95e6-4e73-9421-5b066c089153';

// eslint-disable-next-line camelcase
export const postBody = { mfa_token: 'some_mfa_token', otp: 'ABC123' };
