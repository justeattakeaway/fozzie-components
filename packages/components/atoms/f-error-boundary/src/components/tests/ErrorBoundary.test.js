import { shallowMount, createLocalVue } from '@vue/test-utils';
import ErrorBoundary from '../ErrorBoundary.vue';

const localVue = createLocalVue();

const createComponent = ({
    propsData,
    computed,
    scopedSlots
} = {}) => shallowMount(ErrorBoundary, {
    localVue,
    computed,
    propsData,
    scopedSlots
});

describe('ErrorBoundary', () => {
    const error = 'test error';
    const vm = 'test vm';
    const info = 'test info';
    const tier = 'test tier';
    const loggerPayload = 'test payload';

    describe('render', () => {
        const scopedSlots = {
            default: `<div>
                <p>hasError: {{ props.hasError }}</p>
                <p>error: {{ props.error }}</p>
                <p>vm: {{ props.vm }}</p>
                <p>info: {{ props.info }}</p>
                <p>tier: {{ props.tier }}</p>
                <p>loggerPayload: {{ props.loggerPayload }}</p>
            </div>`
        };

        describe.each([
            [false, false],
            [false, true],
            [true, false],
            [true, true]
        ])('when `hideOnError` is %s and `hasError` is %s', (
            hideOnError,
            hasError
        ) => {
            it('should return expected slot props', async () => {
                // Arrange
                const computed = {
                    loggerPayload: () => loggerPayload
                };
                const propsData = {
                    tier,
                    hideOnError
                };
                const data = {
                    hasError,
                    error,
                    vm,
                    info
                };

                // Act
                const wrapper = createComponent({
                    scopedSlots,
                    computed,
                    propsData
                });

                await wrapper.setData(data);

                // Assert
                expect(wrapper.html()).toMatchSnapshot();
            });
        });
    });

    describe('errorCaptured', () => {
        it('should emit expected payload when error is captured', () => {
            // Arrange
            const computed = {
                loggerPayload: () => loggerPayload
            };
            const propsData = {
                tier
            };

            // Act
            const wrapper = createComponent({
                computed,
                propsData
            });

            wrapper.vm.interceptError(error, vm, info);

            // Assert
            const [[emittedPayload]] = wrapper.emitted('on-error');
            expect(emittedPayload).toEqual({
                error,
                info,
                loggerPayload,
                tier,
                vm
            });
        });
    });
});
