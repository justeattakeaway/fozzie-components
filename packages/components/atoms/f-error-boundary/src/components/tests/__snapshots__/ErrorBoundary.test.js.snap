// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary render when \`hideOnError\` is false and \`hasError\` is false should return expected slot props 1`] = `
<div>
  <p>hasError: false</p>
  <p>error: test error</p>
  <p>vm: test vm</p>
  <p>info: test info</p>
  <p>tier: test tier</p>
  <p>loggerPayload: test payload</p>
</div>
`;

exports[`ErrorBoundary render when \`hideOnError\` is false and \`hasError\` is true should return expected slot props 1`] = `
<div>
  <p>hasError: true</p>
  <p>error: test error</p>
  <p>vm: test vm</p>
  <p>info: test info</p>
  <p>tier: test tier</p>
  <p>loggerPayload: test payload</p>
</div>
`;

exports[`ErrorBoundary render when \`hideOnError\` is true and \`hasError\` is false should return expected slot props 1`] = `
<div>
  <p>hasError: false</p>
  <p>error: test error</p>
  <p>vm: test vm</p>
  <p>info: test info</p>
  <p>tier: test tier</p>
  <p>loggerPayload: test payload</p>
</div>
`;

exports[`ErrorBoundary render when \`hideOnError\` is true and \`hasError\` is true should return expected slot props 1`] = `""`;
